ui/src-tauri/src/lib.rs

```diff
@@ -140,10 +111,46 @@ async fn play_mpv(
     inst: Option<String>,
     sub: Option<String>,
 ) -> Result<(), ()> {
-    let mut app_state = state.lock().await;
-    app_state.playlist.push(PlaylistEntry { video, inst, sub });
+    let app_state = state.lock().await;
+
     if !app_state.mpv_started {
         start_mpv(app_handle, state.inner().clone(), auth);
     }
+
+    tauri::async_runtime::spawn(add_to_mpv_playlist(video, inst, sub));
+
     Ok(())
 }
+
+async fn add_to_mpv_playlist(
+    video: Option<String>,
+    inst: Option<String>,
+    sub: Option<String>,
+) {
+    let mpv = loop {
+        if let Ok(mpv) = Mpv::connect("/tmp/mpv.sock") {
+            break mpv;
+        }
```

Comment:
Don't know if we need to loop (but certainly not forever as it could hide some underlying issue).
And probably should be refactored into its own function.